### DISCUSSION TITLE
DSLR with Deferred Download scripts are broken.

### DISCUSSION THREAD

**Discussion by 3Ddesignbros** [FormerLurker/Octolapse#716] [Q&A]
I followed the deferred download guide and worked perfectly.  Did a couple of timelapse that day with no issues.

I currently have 9 printers and over the last couple days worked on getting them all on Octoprint as well as getting an Octofarm server set up.  Was able to accomplish that without issue.

However, today I tried doing another timelapse but now I get scripts errors.  
All 3 scripts return errors:  External Camer Setup - Script, Before Print Start Script, as well as Before Render Script.

Below are some things I've done since it initially worked form

- Changed the host time of the Octopi from "octopi.local" to "Mini-2.local"
- Set up Octofarm on a separate raspberry pi (this is the only thing on the pi)
- Used a API key with octofarm and this instance of Octoprint.

Couple of things I've tried that didn't work:
- De-linked the instance of octoprint from Octo farm (No effect)
- Reboot Pi (no effect)

Short of that, I don't know.  My gut tells me that changing the hostname could be the problem?  I have no programming knowledge at all but can follow instructions and the youtube videos were great.  But when it comes to troubleshooting I have no programming experience to fall back on.

**Comment by FormerLurker** [comment]
The host name should have nothing to do with it.  These scripts don't interact with the network at all, just your camera (via gphoto2) and the file system.  I'll need some logs to debug this.  Select the debug logging profile and test each script, then download the log (can be done by editing the logging profile), paste to gist.github.com, then drop a link in here.

Also try running the scripts from the terminal (the take-snapshot.sh script is the easiest) and see what happens.

I suspect this issue has something to do with the camera settings.  It's unlikely that all 3 scripts are broken, since they've work for so many people for several years.  It's more likely there is something different about the camera settings.  Knowing what model you are using would help.  Some cameras need to be in a certain mode.  Some cameras will connect by default via UMS drivers (especially on reboot) if the camera is plugged in, making communication via gphoto2 impossible.

**Reply by 3Ddesignbros** [reply]
@FormerLurker Thanks for the super fast reply... I looked at the logs and it said all over it camera not found.  Once I saw that I went in and double-checked all my settings in octolapse.  The camera selected from the drop-down was different from the camera check-marked un Cameras lower down the page.  Corrected that and I'm currently printing and taking pictures!

Thanks.
